### PR TITLE
🧙 Use magefile for build management 

### PR DESCRIPTION
**Pull request by bdeneux**
This pull request introduces a significant refactoring of the build process by incorporating the [Mage](https://magefile.org/) build tool. Mage allows us to write our build scripts in Go, providing a more robust and maintainable approach compared to traditional shell scripts. See #24.

### 👩‍💻 Usage 

Here the list of available command that I've introduced : 

```bash
  build:ts           build typescript schema for the given contract schema.
  schema:clean       remove temporary files
  schema:download    download contracts schemas at a given ref.
  schema:generate    build and generate contracts json schemas at the given ref.
  schema:readme      generate contracts readme on all target
```

This command will download schema located in contracts repository from the main branch and generated appropriate schema into this repository.

```bash
mage schema:generate main
```

### 📋 Next step

Since typescript generation is not used for now and contains some issue on the generated files (#9), I propose to implement the publish action with mage for the next development (#26 )

### CODE REVIEW

File: magefiles/build.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		return err
+	}
+
+	err = sh.Run("yarn", "--cwd", dest)
+	if err != nil {
+		return err
+	}
+
+	return sh.Run("yarn", "--cwd", dest, "build")
+}
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply magefiles/build.go]: Good suggestion, it's done thanks.

File: README.md [hunk 2]
```diff
@@ -17,6 +17,35 @@
...
+mage schema:generate v5.0.0
+```
+
+Build targeted language schema:
+
+```bash
+mage build:ts axone-objectarium
+```
+
 ## You want to get involved? 😍
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply README.md:49]: @coderabbitai No it's what we want.

File: magefiles/schema.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+	for _, schema := range strings.Split(schemas, "\n") {
+		dest := filepath.Join(schema, "../../../", filepath.Base(schema))
+		if err := os.Rename(schema, dest); err != nil {
+			return err
+		}
+	}
+	fmt.Println("✨ Contracts json schema generated")
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply magefiles/schema.go]: @coderabbitai Yes, good catch! It's fixed.

**Review by amimart — APPROVED** [review]
That is really nice thanks! Much more readable this way :)

**Comment by ccamel** [comment]
@bdeneux I took the initiative to apply linting to the magefiles, enhancing their quality (another advantage they have over Makefiles)

**Comment by bdeneux** [comment]
> @bdeneux I took the initiative to apply linting to the magefiles, enhancing their quality (another advantage they have over Makefiles)

@ccamel Wonderful ! I tried to do it but without success since it's specific build and golangci didn't recognize it. Many thanks !!

**Review by ccamel — APPROVED** [review]
Awesome!!! 😍 

mage-small

**Pull request merged**